cypress/integration/tests/controls/stakeholders/interlinked.test.ts

```diff
@@ -0,0 +1,206 @@
+            data.getEmail(),
+            data.getFullName(),
+            "",
+            stakeholdergroupNames
+        );
+        stakeholder.create();
+        cy.wait("@postStakeholder");
+        exists(stakeholder.email);
+
+        // Check if both the stakeholder groups got attached to stakeholder
+        selectItemsPerPage(100);
+        cy.wait(2000);
+        cy.get(tdTag)
+            .contains(stakeholder.email)
+            .parent(trTag)
+            .within(() => {
+                click(expandRow);
+            })
+            .get("div > dd")
+            .should("contain", stakeholdergroupNames[0])
+            .and("contain", stakeholdergroupNames[1]);
+
+        // Update name of second stakeholder group
+        var updatedStakeholderGroupName = data.getCompanyName();
+        stakeholdergroupsList[1].edit({ name: updatedStakeholderGroupName });
+        cy.wait("@getStakeholdergroups");
+        cy.wait(2000);
+
+        // Navigate to stakeholder page
+        clickByText(navTab, stakeholders);
+
+        // Verify if the second stakeholder group's name attached to the stakeholder got updated
+        selectItemsPerPage(100);
+        cy.wait(2000);
+        cy.get(tdTag)
+            .contains(stakeholder.email)
+            .parent(trTag)
+            .within(() => {
+                click(expandRow);
+            })
+            .get("div > dd")
+            .should("contain", updatedStakeholderGroupName);
+
+        // Delete second stakeholder group
+        stakeholdergroupsList[1].delete();
+        cy.wait("@getStakeholdergroups");
+
+        // Assert that second stakeholder group got deleted
+        notExists(stakeholdergroupsList[1].name);
+
+        // Navigate to stakeholder page
+        clickByText(navTab, stakeholders);
+
+        // Verify if the second stakeholder group's name got detached from stakeholder
+        selectItemsPerPage(100);
+        cy.wait(2000);
+        cy.get(tdTag)
+            .contains(stakeholder.email)
+            .parent(trTag)
+            .within(() => {
+                click(expandRow);
+            })
+            .get("div > dd")
+            .should("not.contain", updatedStakeholderGroupName);
+
+        // Delete stakeholder
+        stakeholder.delete();
+        cy.wait("@getStakeholders");
+
+        // Assert that stakeholder got deleted
+        notExists(stakeholder.email);
+
+        // Delete first stakeholder group
+        stakeholdergroupsList[0].delete();
+        cy.wait("@getStakeholdergroups");
+
+        // Assert the deletion of first stakeholder group
+        notExists(stakeholdergroupsList[0].name);
+    });
+
+    it("Job function attach, update and delete dependency on stakeholder", function () {
+        // Create new job function
+        const jobfunction = new Jobfunctions(data.getJobTitle());
+        jobfunction.create();
+        cy.wait("@postJobfunction");
+        exists(jobfunction.name);
```

Comment:
Question: Why jobfunction creation is not added to before block?

Reply:
We can add it, but in this case it doesn't matter since the creation of job function is specific to this test case. It would have been mandatory to create in before block, if we were to use that job function in 1st test case as well

Reply:
Agree then we should not keep stakeholder groups to before block. right?
As it is only used in first test case.

Reply:
Right, that should be the case.